Resources/doc/versioning.rst

```diff
@@ -0,0 +1,180 @@
+
+3 - You must have declared the version value in your config, otherwise it won't be catched :
+
+.. code-block:: yaml
+
+    fos_rest:
+        view:
+            mime_types:
+                json: ['application/json', 'application/json;version=1.0', 'application/json;version=1.1']
+
+Note: If you have to handle huge versions and mime types, you can simplify the configuration with a php script:
+
+.. code-block:: php
+
+    // app/config/fos_rest_mime_types.php
+    $versions = array(
+        '1.0',
+        '1.1',
+        '2.0',
+    );
+
+    $mimeTypes = array(
+        'json' => array(
+            'application/json',
+        ),
+        'yml'  => array(
+            'application/yaml',
+            'text/yaml',
+        ),
+    );
+
+    array_walk($mimeTypes, function (&$mimeTypes, $format, $versions) {
+        $versionMimeTypes = array();
+        foreach ($mimeTypes as $mimeType) {
+            foreach ($versions as $version) {
+                array_push($versionMimeTypes, sprintf('%s;version=%s', $mimeType, $version));
+                array_push($versionMimeTypes, sprintf('%s;v=%s', $mimeType, $version));
+            }
+        }
+        $mimeTypes = array_merge($mimeTypes, $versionMimeTypes);
+    }, $versions);
+
+    $container->loadFromExtension('fos_rest', array(
+        'view' => array(
+            'mime_types' => $mimeTypes,
+        ),
+    ));
+
+And then, import it from your config.yml file:
+
+.. code-block:: yaml
+
+    imports:
+        - { resource: assets_version.php }
+
+If you have to verify if the version is correctly catched you can use something like :
+
+.. code-block:: php
+
+        if ($this->container->get('fos_rest.versioning.listener')) {
+            print $this->container->get('fos_rest.versioning.listener')->getVersion();
+        }
+
+Use the ``JMSSerializer`` with the API versioning
+-------------------------------------------------
+You should have tagged your entities with version information (@Since, @Until ...)
+
+See `this JMS Serializer article`_ for details about versioning objects.
+
+.. _`this JMS Serializer article`: http://jmsyst.com/libs/serializer/master/cookbook/exclusion_strategies#versioning-objects
+
+That's it, it should work now.
+
+How to match a specific version in my routing ?
+-----------------------------------------------
+You can use conditions on your request to check for the version that was determined:
+
+.. code-block:: yaml
+
+    my_route:
+        # ...
+        condition: "request.attributes.get('version') == 'v2'"
+
+When using the :doc:`automatic route generation <5-automatic-route-generation_single-restful-controller>`,
+you can also use the ``@Version`` annotation to set the above condition automatically on all methods
+in the given controller.
+
+.. code-block:: php
+
+    use FOS\RestBundle\Controller\Annotations\Version;
+
+    /**
+     * @Version("v2")
+     *
+     * or if you support multiple versions in this controller
+     * @Version({"v1", "v2"})
```

Comment:
yeah .. I also think it will add confusion .. we have a solution in the docs to make it easier to define multiple mime types quickly .. so yeah lets not bother